### PR TITLE
[DF] Error out when TTreeIndex is used in distributed mode

### PR DESCRIPTION
**Pull request by vepadulano**
We do not support it at the moment, so better to raise an explicit error.

### CODE REVIEW

File: bindings/experimental/distrdf/python/DistRDF/HeadNode.py [hunk 1]
```diff
@@ -377,6 +377,11 @@ def __init__(self, backend: BaseBackend, npartitions: Optional[int], localdf: RO
         if isinstance(args[0], ROOT.TTree):
             # RDataFrame(tree, defaultBranches = {})
             self.tree = args[0]
+            # TTreeIndex is not supported in distributed RDataFrame
+            list_of_friends = self.tree.GetListOfFriends()
+            if list_of_friends and list_of_friends.GetEntries() > 0:
+                if any(friend_element.GetTree().GetTreeIndex() for friend_element in list_of_friends):
+                    raise ValueError("At least one TTreeIndex was detected. This is not supported in distributed mode.")
```
Comment (eguiraud) [inline bindings/experimental/distrdf/python/DistRDF/HeadNode.py:384]: "Friend TTree '{treename}' has a TTreeIndex. This is not supported in distributed mode."

**Review by eguiraud — APPROVED** [review]
LGTM!

**Comment by vepadulano** [comment]
@phsft-bot build

**Comment by vepadulano** [comment]
Test failures do not seem related at first glance. Might need a separate investigation. Rerunning the CI for now

**Comment by vepadulano** [comment]
@phsft-bot build

**Pull request merged**